MvvmCross/Navigation/EventArguments/NavigateEventArgs.cs

```diff
@@ -2,25 +2,32 @@
 // The .NET Foundation licenses this file to you under the MS-PL license.
 // See the LICENSE file in the project root for more information.
 
-using System;
+using System.ComponentModel;
 using System.Threading;
 using MvvmCross.ViewModels;
 
 namespace MvvmCross.Navigation.EventArguments
 {
-    public class NavigateEventArgs : EventArgs
+    public enum NavigationMode
     {
-        public NavigateEventArgs()
+        None,
+        Show,
+        Close
+    }
+
+    public class MvxNavigateEventArgs : MvxCancelEventArgs, IMvxNavigateEventArgs
+    {
+        public MvxNavigateEventArgs(NavigationMode mode, CancellationToken cancellationToken = default) : base(cancellationToken)
         {
+            Mode = mode;
         }
 
-        public NavigateEventArgs(IMvxViewModel viewModel, CancellationToken cancellationToken = default(CancellationToken))
+        public MvxNavigateEventArgs(NavigationMode mode, IMvxViewModel viewModel, CancellationToken cancellationToken = default) : this(mode, cancellationToken)
```

Comment:
I would put the `NavigationMode` after viewmodel, to keep it more like it already was, and to better match the other ctor.